tokio-test/src/macros.rs

```diff
@@ -0,0 +1,109 @@
+//! A collection of useful macros for testing futures and tokio based code
+
+/// Assert if a poll is ready
+#[macro_export]
+macro_rules! assert_ready {
+    ($e:expr) => {{
+        match $e {
+            Ok(::futures::Async::Ready(v)) => v,
+            Ok(_) => panic!("not ready"),
+            Err(e) => panic!("error = {:?}", e),
+        }
+    }};
+    ($e:expr, $($msg:expr),+) => {{
+        match $e {
+            Ok(::futures::Async::Ready(v)) => v,
+            Ok(_) => {
+                let msg = format!($($msg),+);
+                panic!("not ready; {}", msg)
+            }
+            Err(e) => {
+                let msg = format!($($msg),+);
+                panic!("error = {:?}; {}", e, msg)
+            }
+        }
+    }};
+}
+
+/// Asset if the poll is not ready
+#[macro_export]
+macro_rules! assert_not_ready {
+    ($e:expr) => {{
+        match $e {
+            Ok(::futures::Async::NotReady) => {}
+            Ok(::futures::Async::Ready(v)) => panic!("ready; value = {:?}", v),
+            Err(e) => panic!("error = {:?}", e),
+        }
+    }};
+    ($e:expr, $($msg:expr),+) => {{
+        match $e {
+            Ok(::futures::Async::NotReady) => {}
+            Ok(::futures::Async::Ready(v)) => {
+                let msg = format!($($msg),+);
```

Comment:
Not a huge deal, but you can avoid an allocation if you replace this with
```suggestion
                let msg = format_args!($($msg)+);
```

You might need to pass the `format_args!` directly into the `panic!` though; it doesn't always play nice with `let` due to weird lifetime reasons.